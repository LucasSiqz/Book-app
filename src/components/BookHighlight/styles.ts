import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    position: relative;
    display: flex;
    justify-content: center;
    height: 28.2rem;
    background: ${theme.colors.highlight};
    border-bottom-right-radius: 10rem;
  `}
`

export const ArrowButton = styled.button`
  ${({ theme }) => css`
    border: none;
    background: none;
    cursor: pointer;
    height: fit-content;
    padding-top: 5.5rem;
    padding-left: 3.3rem;
    transition: opacity ${theme.transition.default};

    &:hover {
      opacity: 0.7;
    }
  `}
`

export const Image = styled.img`
  position: absolute;
  width: 15.3rem;
  height: 22.9rem;
  filter: drop-shadow(0px 2px 4px rgba(229, 229, 229, 0.5));
  bottom: -3.1rem;
  z-index: 1;

  @media (min-width: 1000px) {
    width: 18.3rem;
    height: 25.9rem;
  }
`

export const WaveContainer = styled.div`
  position: relative;
  margin-left: auto;
  height: 12rem;
  width: 6rem;
  overflow: hidden;
`

export const WaveCircle = styled.img`
  position: absolute;
  top: -3rem;
  right: -5.5rem;
  overflow: hidden;
`

export const WaveCircleSmall = styled.img`
  position: absolute;
  width: 4.8rem;
  height: 4.8rem;
  top: 22.5rem;
  right: 10.2rem;
`

export const BorderCircle = styled.img`
  position: absolute;
  width: 2.4rem;
  height: 2.4rem;
  top: 8.7rem;
  right: 11rem;
`

export const DarkBlueCircle = styled.img`
  position: absolute;
  width: 6.3rem;
  height: 6.3rem;
  top: 10.5rem;
  left: 8rem;
`

export const PinkCircle = styled.img`
  position: absolute;
  width: 1.5rem;
  height: 1.5rem;
  top: 11.5rem;
  left: 5rem;
`
