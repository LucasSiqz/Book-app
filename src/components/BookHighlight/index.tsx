import { useRouter } from 'next/router'

import { BackIcon } from 'components/Icons'

import * as S from './styles'

export type BookHighlightProps = {
  image: string
}

const BookHighlight = ({ image }: BookHighlightProps) => {
  const router = useRouter()

  return (
    <S.Wrapper>
      <S.ArrowButton onClick={() => router.push('/')} title="Go back">
        <BackIcon />
      </S.ArrowButton>
      <S.Image src={image} alt="book cover" />

      {/* Visual Effects */}
      <S.PinkCircle src="../img/pinkCircle.svg" />
      <S.DarkBlueCircle src="../img/darkBlueCircle.svg" />
      <S.BorderCircle src="../img/borderCircle.svg" />
      <S.WaveCircleSmall src="../img/waveCircle.svg" />
      <S.WaveContainer>
        <S.WaveCircle src="../img/waveCircle.svg" />
      </S.WaveContainer>
    </S.Wrapper>
  )
}

export default BookHighlight
