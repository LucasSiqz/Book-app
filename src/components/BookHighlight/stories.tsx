import { Story, Meta } from '@storybook/react/types-6-0'
import BookHighlight, { BookHighlightProps } from '.'

export default {
  title: 'BookHighlight',
  component: BookHighlight,
  args: {
    image: 'img/Book.png'
  }
} as Meta

export const Default: Story<BookHighlightProps> = (args) => (
  <BookHighlight {...args} />
)
