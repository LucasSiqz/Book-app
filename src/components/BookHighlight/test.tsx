import userEvent from '@testing-library/user-event'
import { render, screen } from 'utils/test-utils'

import BookHighlight from '.'

const props = {
  image: '../img/cover.png'
}

// eslint-disable-next-line @typescript-eslint/no-var-requires
const useRouter = jest.spyOn(require('next/router'), 'useRouter')
const push = jest.fn()

useRouter.mockImplementation(() => ({
  push,
  query: '',
  asPath: '',
  route: '/'
}))

describe('<BookHighlight />', () => {
  it('should render correctly', () => {
    const { container } = render(<BookHighlight {...props} />)

    expect(screen.getByRole('img', { name: 'book cover' })).toHaveAttribute(
      'src',
      props.image
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('click on arrow left should redirect to home', () => {
    render(<BookHighlight {...props} />)

    userEvent.click(screen.getByTitle(/Go back/i))
    expect(push).toHaveBeenCalledWith('/')
  })
})
