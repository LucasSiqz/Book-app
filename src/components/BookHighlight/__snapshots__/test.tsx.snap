// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookHighlight /> should render correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 28.2rem;
  background: #FFF6E5;
  border-bottom-right-radius: 10rem;
}

.c1 {
  border: none;
  background: none;
  cursor: pointer;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  margin-top: 5.5rem;
  margin-left: 3.3rem;
  -webkit-transition: opacity 0.3s ease-in-out;
  transition: opacity 0.3s ease-in-out;
}

.c1:hover {
  opacity: 0.7;
}

.c2 {
  position: absolute;
  width: 15.3rem;
  height: 22.9rem;
  -webkit-filter: drop-shadow(0px 2px 4px rgba(229,229,229,0.5));
  filter: drop-shadow(0px 2px 4px rgba(229,229,229,0.5));
  bottom: -3.1rem;
  z-index: 1;
  border-radius: 0.4rem;
}

.c7 {
  position: relative;
  margin-left: auto;
  height: 12rem;
  width: 6rem;
  overflow: hidden;
}

.c8 {
  position: absolute;
  top: -3rem;
  right: -5.5rem;
  overflow: hidden;
}

.c6 {
  position: absolute;
  width: 4.8rem;
  height: 4.8rem;
  top: 22.5rem;
  right: 9.2rem;
}

.c5 {
  position: absolute;
  width: 2.4rem;
  height: 2.4rem;
  top: 8.7rem;
  right: 10.2rem;
}

.c4 {
  position: absolute;
  width: 6.3rem;
  height: 6.3rem;
  top: 10.5rem;
  left: 8rem;
}

.c3 {
  position: absolute;
  width: 1.5rem;
  height: 1.5rem;
  top: 11.5rem;
  left: 5rem;
}

@media (min-width:1000px) {
  .c2 {
    width: 18.3rem;
    height: 25.9rem;
  }
}

<div
  class="c0"
>
  <button
    class="c1"
    title="Go back"
  >
    <svg
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15 8H1"
        stroke="black"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
      <path
        d="M8 15L1 8L8 1"
        stroke="black"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </button>
  <img
    alt="book cover"
    class="c2"
    src="../img/cover.png"
  />
  <img
    class="c3"
    src="../img/pinkCircle.svg"
  />
  <img
    class="c4"
    src="../img/darkBlueCircle.svg"
  />
  <img
    class="c5"
    src="../img/borderCircle.svg"
  />
  <img
    class="c6"
    src="../img/waveCircle.svg"
  />
  <div
    class="c7"
  >
    <img
      class="c8"
      src="../img/waveCircle.svg"
    />
  </div>
</div>
`
